src/shared/components/palette/case-flag/components/manage-case-flags/manage-case-flags.component.ts

```diff
@@ -0,0 +1,109 @@
+import { FormControl, FormGroup } from '@angular/forms';
+import { ErrorMessage } from '../../../../../domain';
+import { CaseFlagState, FlagDetail, Flags } from '../../domain';
+import { CaseFlagFieldState, CaseFlagWizardStepTitle } from '../../enums';
+import { SelectManageCaseFlagErrorMessage } from '../../enums/manage-case-flag-type.enum';
+
+@Component({
+  selector: 'ccd-manage-case-flags',
+  templateUrl: './manage-case-flags.component.html'
+})
+export class ManageCaseFlagsComponent implements OnInit {
+
+  @Input() public formGroup: FormGroup;
+  @Input() public flagsData: Flags[];
+  @Output() public caseFlagStateEmitter: EventEmitter<CaseFlagState> = new EventEmitter<CaseFlagState>();
+
+  public flagLocationTitle: CaseFlagWizardStepTitle;
+  public errorMessages: ErrorMessage[] = [];
+  public manageCaseFlagSelectedErrorMessage: SelectManageCaseFlagErrorMessage = null;
+  public filteredFlagsData: { partyName: string, association: string, comment: string, flagCode: string} [] = [];
+  public caseFlagsConfigError = false;
+  public readonly selectedControlName = 'selectedManageCaseLocation';
+
+  public ngOnInit(): void {
+    this.flagLocationTitle = CaseFlagWizardStepTitle.MANAGE_CASE_FLAGS;
+
+    // Filter out any flags instances that don't have a party name
+    if (this.flagsData) {
+      this.filteredFlagsData = this.flagsData.reduce((flagsData, flagData) => {
+        const result = flagData.details ? flagData.details.map(detail =>  this.flagDetailDisplay(detail, flagData.partyName)) : [];
+        if (result.length) {
+          return result;
+        }
+        return flagsData;
+      }, []) as { partyName: string, association: string, comment: string, flagCode: string }[];
+      console.log('flag-data', this.filteredFlagsData);
+    }
+
+    // Add a FormControl for the selected case flag if there is at least one flags instance remaining after filtering
+    if (this.filteredFlagsData && this.filteredFlagsData.length > 0) {
+      this.formGroup.addControl(this.selectedControlName, new FormControl(null));
+    } else {
+      // No filtered flags instances mean there are no parties to select from. The case has not been configured properly
+      // for case flags and the user cannot proceed with flag creation. (Will need to be extended to check for case-level
+      // flags in future)
+      this.onCaseFlagsConfigError();
+    }
+  }
+
+  public flagDetailDisplay(flagDetail: FlagDetail, partyName):
+  { partyName: string, association: string, comment: string, flagCode: string } {
+    return {
+      partyName,
+      association: flagDetail.name ? flagDetail.name : '',
+      comment: flagDetail.flagComment ? '(' + flagDetail.flagComment + ')' : '',
+      flagCode: flagDetail.flagCode ?  flagDetail.flagCode : '',
+    };
+  }
+
+  public processLabel(flagDetail: { partyName: string, association: string, comment: string, flagCode }): string {
```

Comment:
@tarikdogan-hmcts - a unit test should be added for this function to check it is returning the expected string